modules/container/src/main/java/org/mule/runtime/container/internal/TestContainerClassLoaderFactory.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.container.internal;
+
+import static java.util.Collections.emptyMap;
+import static org.mule.runtime.core.api.util.ClassUtils.withContextClassLoader;
+
+import org.mule.runtime.container.api.ModuleRepository;
+import org.mule.runtime.container.api.MuleModule;
+import org.mule.runtime.module.artifact.api.classloader.ArtifactClassLoader;
+import org.mule.runtime.module.artifact.api.classloader.ClassLoaderLookupPolicy;
+import org.mule.runtime.module.artifact.api.classloader.MuleArtifactClassLoader;
+import org.mule.runtime.module.artifact.api.descriptor.ArtifactDescriptor;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * Extends the default {@link ContainerClassLoaderFactory} for testing in order to add boot packages and build a
+ * {@link ClassLoader} for the container that do resolves classes instead of delegating to its parent and also allows to create
+ * the container {@link ClassLoaderLookupPolicy} based on a {@link ClassLoader}.
+ *
+ * @since 4.6
+ */
+public class TestContainerClassLoaderFactory extends ContainerClassLoaderFactory implements AutoCloseable {
```

Comment:
can we configure the behaviour on the ContainerClassLoaderFactory instead of subclassing it?

Reply:
This child class modifies basically everything about the creation of the container class loader. It's not possible to parameterize it and embed it in the parent class.